snapshot_manager/snapshot_manager/config.py

```diff
@@ -113,3 +117,78 @@ def yyyymmdd(self) -> str:
         '20240229'
         """
         return self.datetime.strftime("%Y%m%d")
+
+    def to_github_dict(self) -> dict:
+        """Returns a subset of config entries to be used in a github workflow matrix.
+
+        The keys in this dict are accessed from github workflow files using the "matrix." object.
+        For example the maintainer handle will be accessed as `${{ matrix.maintainer_handle }}`.
+
+        Examples:
+
+        >>> import pprint
+        >>> pprint.pprint(Config(build_strategy="mybuildstrategy",
+        ...   copr_target_project="@mycoprgroup/mycoprproject",
+        ...   package_clone_url="https://src.fedoraproject.org/rpms/mypackage.git",
+        ...   package_clone_ref="mainbranch",
+        ...   maintainer_handle="fakeperson",
+        ...   copr_project_tpl="SomeProjectTemplate-YYYYMMDD",
+        ...   copr_monitor_tpl="https://copr.fedorainfracloud.org/coprs/g/mycoprgroup/SomeProjectTemplate-YYYYMMDD/monitor/",
+        ...   chroot_pattern="^(fedora-(rawhide|[0-9]+)|rhel-[8,9]-)",
+        ...   chroots=["fedora-rawhide-x86_64", "rhel-9-ppc64le"]
+        ... ).to_github_dict())
+        {'chroot_pattern': '^(fedora-(rawhide|[0-9]+)|rhel-[8,9]-)',
+         'chroots': 'fedora-rawhide-x86_64 rhel-9-ppc64le',
+         'clone_ref': 'mainbranch',
+         'clone_url': 'https://src.fedoraproject.org/rpms/mypackage.git',
+         'copr_monitor_tpl': 'https://copr.fedorainfracloud.org/coprs/g/mycoprgroup/SomeProjectTemplate-YYYYMMDD/monitor/',
+         'copr_ownername': '@fedora-llvm-team',
+         'copr_project_tpl': 'SomeProjectTemplate-YYYYMMDD',
+         'copr_target_project': '@mycoprgroup/mycoprproject',
+         'maintainer_handle': 'fakeperson',
+         'name': 'mybuildstrategy'}
+        """
+        return {
+            "name": self.build_strategy,
+            "copr_target_project": self.copr_target_project,
+            "clone_url": self.package_clone_url,
+            "clone_ref": self.package_clone_ref,
+            "maintainer_handle": self.maintainer_handle,
+            "copr_ownername": self.copr_ownername,
+            "copr_project_tpl": self.copr_project_tpl,
+            "copr_monitor_tpl": self.copr_monitor_tpl,
+            "chroot_pattern": self.chroot_pattern,
+            "chroots": " ".join(self.chroots),
+        }
+
+
+def build_config_map() -> dict[str, Config]:
+    """Builds a dictionary for each supported build strategy with the name of the build strategy as key.
+
+    Returns:
+        dict: The config map with build strategies as keys and config objects as values.
+    """
+    configs = [
+        Config(
+            build_strategy="big-merge",
+            copr_target_project="@fedora-llvm-team/llvm-snapshots",
+            package_clone_url="https://src.fedoraproject.org/rpms/llvm.git",
+            package_clone_ref="rawhide",
+            maintainer_handle="tuliom",
```

Comment:
This is out-of-date.

Reply:
Let me fix that.

Reply:
Fixed in 2c267f8c2438650560321c71eddd229915184c01